### PR TITLE
doc: Adds build pipeline documentation

### PR DESCRIPTION
**Pull request by benventive**
GitHub Issue: #

## PR Type

What kind of change does this PR introduce?

- Documentation content changes

## Description

Adds a detailed description of the build definition.

## PR Checklist 
Please check if your PR fulfills the following requirements:

- [ ] Interface members are XML documented
- [ ] Documentation (XML or comments) has been added and/or existing documentation has been updated
- [ ] [Architecture documents](./doc/Architecture.md) have been updated
- [ ] Tested on all relevant platforms

### CODE REVIEW

**Review by YGuerin — APPROVED** [review]

**Review by Guidemarcus — APPROVED** [review]

File: doc/BuildPipeline.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  3. Release the production application to **AppCenter**, the **Google Play Store** and the **Apple AppStore** (the last two are usually configured to be run manually using [approvals](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/approvals?view=azure-devops&tabs=check-pass#approvals))
+
+## Pull request builds
+Due to the length of Xamarin build, this template is configured to behave a little differently when building a pull request (PR). We developed [a set of tools](https://github.com/nventive/MSBuild.UnifiedExtensions/blob/develop/src/Application.Building.Light/Readme.md) to reduce the build time at the cost of some optimizations. This requires a specific variable called `IsLightBuild` to be set, hence why it is appearing in the pipeline. 
+
+Also, being a PR build, we do not find it necessary to release it, so all the relevant stages are disabled.
+
+## Canary
+This template also includes build and release stages dedicated for Canary testing our packages. For the build side, the same templates as the other builds are used and we simply pass the required steps through variables. The release process is identical: we only release in AppCenter.
+
```
Comment (MatFillion) [inline doc/BuildPipeline.md:16]: What do you think of this? Is it necessary or it's redundant?
```suggestion
This step is only recommended for some projects at nventive, feel free to remove those from the yml if you don't need them but it won't impact anything if you leave the yml code.
```

File: doc/BuildPipeline.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+# Build
+## [stage-build.yml](../build/stage-build.yml)
+This file is the template used for our different build stages. Due to the constraints of some of the technologies in our stack - namely, UAP and iOS- this stage is split in two jobs: 
+- One building Android and UAP and running on Windows
+- One building iOS running on macOS
+
+Those 2 jobs rely on the same template for the actual build steps.
+
+**Note**: The WASM portion of this template is not fully functional yet, but it would be running on Linux.
```
Comment (MatFillion) [inline doc/BuildPipeline.md:25]: We should probably remove this and add into an issue instead. I'm afraid when we will do the wasm we will forget to update this :)
You can add right away that the wasm is building on linux maybe?

File: doc/BuildPipeline.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+## [steps-build.yml](../build/steps-build.yml)
+This is where the exact build steps are defined. The sequence is fairly classic for a .Net application:
+1. Install and run [GitVersion](https://gitversion.net/) to calculate the semantic version based on the Git history
+1. Install and run NuGet to install the project's dependencies
+1. Depending on the platform, install the proper signing certificates
+1. Run the build with MSBuild
+1. Run Unit Test and publish both the test results and the code coverage results 
+1. Push the built artifcats
```
Comment (MatFillion) [inline doc/BuildPipeline.md:34]:
```suggestion
1. Push the built artifacts
```

File: doc/BuildPipeline.md (hunk 2 above)
Comment (benventive) [reply doc/BuildPipeline.md:25]: Created #58 
As for the rest, I would leave it like that and, when the WASM build is fixed, this documentation should be updated.

File: doc/BuildPipeline.md (hunk 1 above)
Comment (benventive) [reply doc/BuildPipeline.md:16]: Thanks! I've rephrased a bit and added it!

**Pull request merged**